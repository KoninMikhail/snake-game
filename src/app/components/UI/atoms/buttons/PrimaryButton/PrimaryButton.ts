import styled from 'styled-components';

export const PrimaryButton = styled.button`
    cursor: pointer;
    display: block;
    color: ${({ theme }) => theme.colors.primary};
    border: 3px dashed ${({ theme }) => theme.colors.primary};
    padding: 5px 25px;
    background: black;
    font-family: ${({ theme }) => theme.typography.fontFamily.body};
    text-align: center;
    font-size: ${({ theme }) => theme.typography.fontSize.body};
    font-weight: normal;

    &:hover {
        color: ${({ theme }) => theme.colors.body};
        background: ${({ theme }) => theme.colors.primary};
        border: 3px dashed ${({ theme }) => theme.colors.black};
    }
`;
