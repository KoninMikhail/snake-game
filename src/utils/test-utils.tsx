import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
