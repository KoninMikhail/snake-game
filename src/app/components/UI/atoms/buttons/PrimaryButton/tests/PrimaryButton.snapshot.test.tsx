import { PrimaryButton } from '../PrimaryButton';
import { renderWithProviders } from '@utils/test-utils';

/**
 * Test
 */
describe('PrimaryButton', () => {
    it('equal snapshot', () => {
        const { container } = renderWithProviders(<PrimaryButton>Hello world</PrimaryButton>);
        expect(container).toMatchSnapshot();
    });
});
