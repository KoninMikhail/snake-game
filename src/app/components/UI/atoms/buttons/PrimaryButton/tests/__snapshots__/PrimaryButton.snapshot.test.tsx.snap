// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton equal snapshot 1`] = `
.c0 {
  cursor: pointer;
  display: block;
  color: deeppink;
  border: 3px dashed deeppink;
  padding: 5px 25px;
  background: black;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  text-align: center;
  font-size: small:14px;
  middle: 16px;
  large: 1.1vw;
  font-weight: normal;
}

.c0:hover {
  color: white;
  background: deeppink;
  border: 3px dashed black;
}

<div>
  <button
    class="c0"
  >
    Hello world
  </button>
</div>
`;
